java/core/src/main/java/com/google/protobuf/Utf8.java

```diff
@@ -614,8 +615,35 @@ private static int partialIsValidUtf8(final ByteBuffer buffer, int index, final
      *
      * @throws InvalidProtocolBufferException if the byte array slice is not valid UTF-8.
      */
-    abstract String decodeUtf8(byte[] bytes, int index, int size)
-        throws InvalidProtocolBufferException;
+    String decodeUtf8(byte[] bytes, int index, int size)
+        throws InvalidProtocolBufferException {
+      final String s;
+      try {
+        s = new String(bytes, index, size, StandardCharsets.UTF_8);
```

Comment:
we can't use StandardCharsets yet due to android compatiblity

Reply:
Sure